api-report/core-interfaces.api.md

```diff
@@ -231,10 +239,10 @@ export interface ITelemetryGenericEvent extends ITelemetryProperties {
 
 // @public
 export interface ITelemetryLogger extends ITelemetryBaseLogger {
-    send(event: ITelemetryBaseEvent): void;
-    sendErrorEvent(event: ITelemetryErrorEvent, error?: any): void;
-    sendPerformanceEvent(event: ITelemetryPerformanceEvent, error?: any): void;
-    sendTelemetryEvent(event: ITelemetryGenericEvent, error?: any): void;
+    send(event: ITelemetryBaseEvent, logLevel?: LogLevel): void;
+    sendErrorEvent(event: ITelemetryErrorEvent, error?: any, logLevel?: LogLevel): void;
```

Comment:
i don't understand what the log levels mean or how they intersect with categories. For instance, how is error log level different from error category?

What is the difference between error and warning?

Overall, i think category and level are very similar, i'm wondering if we should just add add a verbose category. If not, i would like to see the full matrix of categories and level specified, what each means, and when to use it.

Reply:
The way I think is people can log many perf/generic events which are verbose and are not logged by default maybe in case of debugging. So, for example we may want to have all features of a perf event but log level is still verbose rather than implementing another sendVerboseEvent() as this event could be a perf event or a generic event.
Here is the table that I am thinking. Added comments in code too. Can remove "Warning" for now as it is not used and can be added later on. Was just taking it from browser log verboseness levels.
<p>&nbsp;</p><figure class="table">

Category/LogLevel | Verbose | Default | Warning | Error
-- | -- | -- | -- | --
Performance | 1 | 1 | x | x
Generic | 1 | 1 | 1 | x
Error | x | x | x | 1

</figure><p>&nbsp;</p>